analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -338,9 +338,12 @@ class GoMod(
  * A class to represent a graph with dependencies. This representation is basically an adjacency list implemented by a
  * map whose keys are package identifiers and whose values are the identifiers of packages these packages depend on.
  */
-private class Graph(private val nodeMap: MutableMap<Identifier, Set<Identifier>> = mutableMapOf()) {
+private class Graph(nodeMap: MutableMap<Identifier, Set<Identifier>> = mutableMapOf()) {
     companion object : Logging
 
+    private val nodeMap: MutableMap<Identifier, MutableSet<Identifier>> =
+        nodeMap.mapValuesTo(mutableMapOf()) { it.value.toMutableSet() }
```

Comment:
How about using `MutableSet<Identifier>` in the constructor instead? All but one caller already use a `MutableSet` anyway, so you only need to adjust the call in `breakCycles()` from

    return Graph(outgoingEdgesForNodes.mapValuesTo(mutableMapOf()) { it.value.toSet() })

to

    return Graph(outgoingEdgesForNodes.mapValuesTo(mutableMapOf()) { it.value.toMutableSet() })